src/gt4py/gtc/python/npir_gen.py

```diff
@@ -0,0 +1,128 @@
+from eve.codegen import FormatTemplate, JinjaTemplate, TemplatedGenerator
+
+from gt4py.gtc import common
+
+
+__all__ = ["NpirGen"]
+
+COMPUTATION_JTPL = """
+def run({{ signature }}):
+    {{ domain_padding | indent(4) }}
+    {{ data_views | indent(4) }}
+    {% for pass in vertical_passes %}\
+    {{ pass | indent(4) }}
+    {% endfor %}\
+"""
+
+
+class NpirGen(TemplatedGenerator):
+
+    Literal = FormatTemplate("np.{_this_node.dtype.name.lower()}({value})")
+
+    def visit_NumericalOffset(self, node, **kwargs):
+        operator = ""
+        delta = node.value
+        if node.value == 0:
+            delta = ""
+        elif node.value < 0:
+            operator = " - "
+            delta = abs(node.value)
+        else:
+            operator = " + "
+        return self.generic_visit(node, op=operator, delta=delta, **kwargs)
+
+    NumericalOffset = FormatTemplate("{op}{delta}")
+
+    def visit_AxisOffset(self, node, **kwargs):
+        offset = self.visit(node.offset)
+        axis_name = self.visit(node.axis_name)
+        if node.parallel:
+            parts = [f"{axis_name.lower()}{offset}", f"{axis_name.upper()}{offset}"]
+            if node.offset.value != 0:
+                parts = [f"({part})" for part in parts]
+            return self.generic_visit(node, from_visitor=":".join(parts), **kwargs)
+        return self.generic_visit(node, from_visitor=f"{axis_name.lower()}_{offset}", **kwargs)
+
+    AxisOffset = FormatTemplate("{from_visitor}")
+
+    FieldSlice = FormatTemplate("{name}_[{i_offset}, {j_offset}, {k_offset}]")
+
+    VectorAssign = FormatTemplate("{left} = {right}")
+
+    VectorArithmetic = FormatTemplate("({left} {op} {right})")
+
+    def visit_VerticalPass(self, node, **kwargs):
+        for_loop_line = ""
+        body_indent = 0
+        if node.direction == common.LoopOrder.FORWARD:
+            for_loop_line = "\nfor k_ in range(k, K):"
+            body_indent = 4
+        elif node.direction == common.LoopOrder.BACKWARD:
+            for_loop_line = "\nfor k_ in range(K, k, -1):"
+            body_indent = 4
+        return self.generic_visit(
+            node, for_loop_line=for_loop_line, body_indent=body_indent, **kwargs
+        )
+
+    VerticalPass = JinjaTemplate(
```

Comment:
I prefer the initial out of class template over this. This is doing magic for now reason and if for whatever reason you move this piece of code around it stops working.

My personal preference is:
- multi line strings
- out of class definition as you had it initially
- list of strings
- this

I can live with the out of class definition.